firmware/application/ui_external_items_menu_loader.cpp

```diff
@@ -0,0 +1,137 @@
+#include "ui_external_items_menu_loader.hpp"
+
+#include "sd_card.hpp"
+
+namespace ui {
+
+std::vector<Bitmap*> ExternalItemsMenuLoader::bitmaps;
+
+std::vector<GridItem> ExternalItemsMenuLoader::load_external_items(app_location_t app_location, NavigationView& nav) {
+    for (const auto& bitmap : bitmaps) {
+        delete bitmap->data;
+        delete bitmap;
+    }
+    bitmaps.clear();
+
+    std::vector<GridItem> exteral_apps;
+
+    if (sd_card::status() != sd_card::Status::Mounted)
+        return exteral_apps;
+
+    for (const auto& entry : std::filesystem::directory_iterator(u"APPS", u"*.ppma")) {
+        FIL ppma_file;
+        auto filePath = u"/APPS/" + entry.path().native();
+        if (f_open(&ppma_file, reinterpret_cast<const TCHAR*>(filePath.c_str()), FA_READ) != FR_OK)
+            continue;
+
+        size_t bytes_read;
+        application_information_t application_information = {};
+        if (f_read(&ppma_file, &application_information, sizeof(application_information_t), &bytes_read) != FR_OK) continue;
+
+        if (application_information.menu_location != app_location)
```

Comment:
Not important for a first version, but I wonder if caching the app info would be a good idea as the number of apps grows.

Reply:
Lets do this, when it becomes a problem.